tests/e2e/subgraph/check.rs

```diff
@@ -0,0 +1,50 @@
+use std::path::PathBuf;
+use std::process::Command;
+
+use assert_cmd::prelude::CommandCargoExt;
+use regex::Regex;
+use rstest::rstest;
+use speculoos::assert_that;
+use speculoos::boolean::BooleanAssertions;
+use tracing_test::traced_test;
+
+use crate::e2e::{remote_supergraph_graphref, test_artifacts_directory};
+
+#[rstest]
+#[ignore]
+#[tokio::test(flavor = "multi_thread")]
+#[traced_test]
+async fn e2e_test_rover_subgraph_check(
+    remote_supergraph_graphref: String,
+    test_artifacts_directory: PathBuf,
+) {
+    // GIVEN
+    //   - a schema that would break the remote schema
+    let subgraph_schema = test_artifacts_directory.join("subgraph/inventory.graphql");
+    let subgraph_schema = subgraph_schema
+        .to_str()
+        .expect("Couldn't get subgraph schema for graph check");
+
+    // WHEN
+    //   - the command is run
+    let mut cmd = Command::cargo_bin("rover").expect("Could not find necessary binary");
+    cmd.args([
+        "subgraph",
+        "check",
+        "--name",
+        "inventory",
+        "--schema",
+        subgraph_schema,
+        &remote_supergraph_graphref,
+    ]);
+    let output = cmd.output().expect("Could not run command");
+
+    // THEN
+    //   - it matches a regex showing that it linted and passed, meaning a check was performed
+    //   against the remote schema using the local one
+    let stdout = std::str::from_utf8(&output.stdout).expect("failed to convert bytes to a str");
+    let re = Regex::new(r"Linter Check \[PASSED\]").unwrap();
```

Comment:
I wonder if checking against the output of log messages is the best plan here? Would it be better to assert on the structure of the response or the exit code?